lib/base/scriptutils.cpp

```diff
@@ -453,7 +453,7 @@ Array::Ptr ScriptUtils::GetTemplates(const Type::Ptr& type)
 
 	ArrayData result;
 
-	for (const ConfigItem::Ptr& item : ConfigItem::GetItems(type)) {
+	for (auto& item : *ConfigItem::GetItems(type)) {
```

Comment:
@julianbrost Do I remember correctly that this shared pointer instance shall stay alive for the whole for loop?

Reply:
https://en.cppreference.com/w/cpp/language/range-for#Temporary_range_expression (for standards before C++23):
> Lifetimes of all temporaries within range-expression are not extended

See also the following example:

```c++
#include <vector>
#include <memory>
#include <iostream>

using namespace std;

shared_ptr<vector<int>> f() {
	return {new vector<int>{4}, [](vector<int> *p) { cout << "delete" << endl; delete p; }};
}

int main() {
	for (auto &x : *f()) {
		cout << "loop" << endl;
		break;
	}
}
```

Produces the following output for me (had to add the `break` because it generated an endless loop for me otherwise, undefined behavior is fun):

```
delete
loop
```

Fun fact: that happened even with `g++ -std=c++23`, but well, that's not even fully standardized, so whatever.

Reply:
So, TL;DR, I **have to** make a temp. var.?

Reply:
Yes